### DISCUSSION TITLE
Doing something wrong, help?

### DISCUSSION THREAD

**Discussion by QuantumDonDoerner** [apache/airflow#14362] [Q&A] [answered]
Trying to quickstart Airflow on Docker.

- Well maintained Ubuntu 20.04 LTS system
- Docker version 19.03.8, build afacb8b7f0
- docker-compose version 1.25.0, build unknown

Following instructions from http://airflow.apache.org/docs/apache-airflow/stable/start/docker.html

- Incanted **docker-compose up airflow-init**, got...

ERROR: The Compose file './docker-compose.yaml' is invalid because:
Invalid top-level property "x-airflow-common". Valid top-level sections for this Compose file are: version, services, networks, volumes, and extensions starting with "x-".

You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version (e.g "2.2" or "3.3") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/
services.airflow-init.depends_on contains an invalid type, it should be an array
services.airflow-scheduler.depends_on contains an invalid type, it should be an array
services.airflow-webserver.depends_on contains an invalid type, it should be an array
services.airflow-worker.depends_on contains an invalid type, it should be an array
services.flower.depends_on contains an invalid type, it should be an array

Any idea what I am doing wrong?

**Comment by turbaszek** [comment, marked as answer]
In my case bumping docker-compose version helped: `version: '3.9'`. Also you may need to update docker-compose:
> Install Docker Compose v1.27.0 and newer on your workstation.

**Reply by QuantumDonDoerner** [reply]
Yep, that did it, thank you very much - and for what it's worth, both changes were needed.

For others trying to bring up airflow on Ubuntu 20.04 LTS, the docker-compose that you get from a naïve "apt-get install" is 1.25.0 - which won't work.  I uninstalled that and installed a more up-to-date docker-compose using pip.  After that, all went well.

**Reply by kamilliano** [reply]
I just bumped into this. I tried explicitly changing the version tag to 3.3 and 2.2, the latter worked on `docker-compose version 1.22.0, build f46880fe.` So either try different APIs or bump docker-compose

**Comment by eff917** [comment]
In it's current form the compose file is "broken".
From the docker-compose documentation:
> Version 3 no longer supports the condition form of depends_on.

https://docs.docker.com/compose/compose-file/compose-file-v3/
I went with the following modifications:
`version: '3.8'`
and 
`  depends_on:
    - redis
    - postgres
`

The other solution is probably to change version to `2.4`, but I haven't tried it.

**Reply by PsycheShaman** [reply]
changing to 2.4 and installing a more up to date docker compose using pip worked for me

**Comment by polmonso** [comment]
At the current airflow release (2.2.2) we had to install docker-compose 1.29.1 as commented on the doc, otherwise the syntax gave errors.

```
Install Docker Compose v1.29.1 and newer on your workstation.
```

https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html

**Comment by maaydi** [comment]
just install last version of `docker compose` 1.29.2 
https://docs.docker.com/compose/install/
